serving/src/test/java/feast/serving/controller/ServingServiceGRpcControllerTest.java

```diff
@@ -59,23 +76,66 @@ public void setUp() {
                     .putFields("entity1", Value.newBuilder().setInt64Val(1).build())
                     .putFields("entity2", Value.newBuilder().setInt64Val(1).build()))
             .build();
+  }
 
+  private ServingServiceGRpcController getServingServiceGRpcController(boolean enableAuth) {
     Tracer tracer = Configuration.fromEnv("dummy").getTracer();
     FeastProperties feastProperties = new FeastProperties();
-    service = new ServingServiceGRpcController(mockServingService, feastProperties, tracer);
+
+    AuthorizationProperties authorizationProps = new AuthorizationProperties();
+    authorizationProps.setEnabled(enableAuth);
+    AuthenticationProperties authenticationProps = new AuthenticationProperties();
+    authenticationProps.setEnabled(enableAuth);
+    SecurityProperties securityProperties = new SecurityProperties();
+    securityProperties.setAuthentication(authenticationProps);
+    securityProperties.setAuthorization(authorizationProps);
+    feastProperties.setSecurity(securityProperties);
+    AuthorizationService authorizationservice =
+        new AuthorizationService(feastProperties.getSecurity(), authProvider);
+    return new ServingServiceGRpcController(
+        authorizationservice, mockServingService, feastProperties, tracer);
   }
 
   @Test
   public void shouldPassValidRequestAsIs() {
+    service = getServingServiceGRpcController(false);
     service.getOnlineFeatures(validRequest, mockStreamObserver);
     Mockito.verify(mockServingService).getOnlineFeatures(validRequest);
   }
 
   @Test
   public void shouldCallOnErrorIfEntityDatasetIsNotSet() {
+    service = getServingServiceGRpcController(false);
     GetOnlineFeaturesRequest missingEntityName =
         GetOnlineFeaturesRequest.newBuilder(validRequest).clearEntityRows().build();
     service.getOnlineFeatures(missingEntityName, mockStreamObserver);
     Mockito.verify(mockStreamObserver).onError(Mockito.any(StatusRuntimeException.class));
   }
+
+  @Test
+  public void shouldPassValidRequestAsIsIfRequestIsAuthorized() {
+    service = getServingServiceGRpcController(true);
+    Authentication auth = mock(Authentication.class);
+    SecurityContext context = mock(SecurityContext.class);
+    SecurityContextHolder.setContext(context);
+    when(context.getAuthentication()).thenReturn(auth);
+    doReturn(AuthorizationResult.success())
+        .when(authProvider)
+        .checkAccessToProject(anyString(), any(Authentication.class));
+    service.getOnlineFeatures(validRequest, mockStreamObserver);
+    Mockito.verify(mockServingService).getOnlineFeatures(validRequest);
+  }
+
+  @Test(expected = AccessDeniedException.class)
+  public void shouldThrowErrorOnValidRequestIfRequestIsUnauthorized() {
+    service = getServingServiceGRpcController(true);
+    Authentication auth = mock(Authentication.class);
```

Comment:
Nitpick but can we move the minor duplication here up to the `setUp()` method or a common method?